shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -263,49 +368,73 @@ private void setSystemChromeSystemUIOverlayStyle(
     Window window = activity.getWindow();
     View view = window.getDecorView();
     int flags = view.getSystemUiVisibility();
-    // You can change the navigation bar color (including translucent colors)
-    // in Android, but you can't change the color of the navigation buttons until
-    // Android O.
-    // LIGHT vs DARK effectively isn't supported until then.
-    // Build.VERSION_CODES.O
-    if (Build.VERSION.SDK_INT >= 26) {
-      if (systemChromeStyle.systemNavigationBarIconBrightness != null) {
-        switch (systemChromeStyle.systemNavigationBarIconBrightness) {
-          case DARK:
-            // View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
-            flags |= 0x10;
-            break;
-          case LIGHT:
-            flags &= ~0x10;
-            break;
-        }
-      }
-      if (systemChromeStyle.systemNavigationBarColor != null) {
-        window.setNavigationBarColor(systemChromeStyle.systemNavigationBarColor);
-      }
+
+    // SYSTEM STATUS BAR -------------------------------------------------------------------
+    // You can't change the color of the system status bar until SDK 21.
+    // If transparent, SDK 29 and higher may apply a translucent scrim behind the bar to ensure
+    // proper contrast. This can be overridden with
+    // SystemChromeStyle.systemStatusBarContrastEnforced.
+    if (systemChromeStyle.statusBarColor != null
+        && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+      window.setStatusBarColor(systemChromeStyle.statusBarColor);
     }
-    // Build.VERSION_CODES.M
-    if (Build.VERSION.SDK_INT >= 23) {
-      if (systemChromeStyle.statusBarIconBrightness != null) {
-        switch (systemChromeStyle.statusBarIconBrightness) {
-          case DARK:
-            // View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
-            flags |= 0x2000;
-            break;
-          case LIGHT:
-            flags &= ~0x2000;
-            break;
-        }
+    // You can't change the color of the status icons until SDK 23.
+    if (systemChromeStyle.statusBarIconBrightness != null && Build.VERSION.SDK_INT >= 23) {
+      switch (systemChromeStyle.statusBarIconBrightness) {
+        case DARK:
+          // View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
```

Comment:
This flag appears deprecated in Android 30.

Reply:
Yes, the intention of this change is not to migrate to new api at this time. I just refactored to fix a bug.